src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Entry[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
 
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        for (Entry<K, V> entry = table[index]; entry != null; entry = entry.next) {
+            if ((entry.hash == hash) && (key == entry.key || key.equals(entry.key))) {
+                entry.value = value; // Update the value if key already exists
+                return;
+            }
+        }
+        addEntry(hash, key, value, index);
```

Comment:
any reason for that? you should do all the `add` only inside `put` method. refactor code accordingly
```suggestion
```